cstar/entrypoint/worker/worker.py

```
@@ -0,0 +1,479 @@
+            self.log.debug(f"Removing existing output dir: {self._output_root}")
+            shutil.rmtree(self._output_root)
+
+        # leftover external code folder causes non-empty repo errors; remove.
+        externals_path = cstar_sysmgr.environment.package_root / "externals"
+        if externals_path.exists():
+            self.log.debug(f"Removing existing externals dir: {externals_path}")
+            shutil.rmtree(externals_path)
+        externals_path.mkdir(parents=True, exist_ok=False)
+
+        # create a clean location to write outputs.
+        if not self._output_dir.exists():
+            self.log.debug(f"Creating clean output dir: {self._output_dir}")
+            self._output_dir.mkdir(parents=True, exist_ok=True)
+
+    def _log_disposition(self) -> None:
+        """Log the status of the simulation at shutdown time."""
+        disposition: ExecutionStatus = (
+            self._handler.status if self._handler else ExecutionStatus.UNKNOWN
+        )
+
+        if self._handler:
+            self.log.info(f"Completed simulation logs at: {self._handler.output_file}")
+
+        if disposition == ExecutionStatus.COMPLETED:
+            self.log.info("Simulation completed successfully.")
+        elif disposition == ExecutionStatus.FAILED:
+            self.log.error("Simulation failed.")
+        else:
+            self.log.warning(f"Simulation ended with status: {disposition}")
+
+    @override
+    def _on_start(self) -> None:
+        """Prepare the simulation for execution.
+
+        Verifies the simulation loaded properly, configuring the file system, retrieving
+        remote resources, and building third-party codebases.
+        """
+        self._prepare_file_system()
+
+        if self._blueprint_uri is None:
+            raise BlueprintError("No blueprint URI provided")
+
+        if self._simulation is None:
+            raise BlueprintError(f"Unable to load the blueprint: {self._blueprint_uri}")
```

Might as well fail faster on these, either before `_prepare_file_system`, or even put them in init? Unless we expect post-init modifications, which seems sus.

---

failing faster!